sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -318,16 +318,28 @@ case class AlterTableChangeColumnCommand(
           s"'${newColumn.name}' with type '${newColumn.dataType}'")
     }
 
+    val changeSchema = originColumn.dataType != newColumn.dataType
     val newSchema = table.schema.fields.map { field =>
       if (field.name == originColumn.name) {
-        // Create a new column from the origin column with the new comment.
-        addComment(field, newColumn.getComment)
+        var newField = field
```

Comment:
I'd recommend getting rid of this `var` and re-writting the code as follows:

```
val newField = newColumn.getComment.map(...).getOrElse(field)
```

Reply:
More clear for getting rid of var, pls check next patch. If we implement rename or others meta change feature here, may still need some code rework.